### PR TITLE
Preserve data

### PR DESCRIPTION
**Pull request by abiola-arobieke**
In this milestone:

The books.json, rentals.json, and person.json files were added to the project.
Data (books, person, and rentals) were saved on exit from the app into the JSON files.
On startup, data were loaded from the JSON files.

### CODE REVIEW

**Review by tresorsawasawa — COMMENTED** [review]
# STATUS: INVALID CODE REVIEW REQUEST 🛑 🛑

Hi @abiola-arobieke and @flemton,:wave: :wave: 
While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request 🛑 🛑

You have submitted a project that doesn't run when trying to test it. Make sure to fix the below error before requesting a new review.
Screenshot from 2023-07-13 20-12-39

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

---
Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

---
_Invalid Code Review Request **Does not count** into the code reviews limit._

File: json/person.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  },
+  {
+    "type": "teacher",
+    "id": 531,
+    "age": 39,
+    "specialization": "Maths",
+    "name": "Mrs Oluyemi",
+    "parent_permission": true
+  }
+]
```
Comment (Ibizugbe) [inline json/person.json]: Nice job team👍. Currently, when trying to add a person, an error is thrown to the user preventing the person from being added and the data from being preserved👍.

image

File: json/books.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+[
+  {
+    "title": "The Moonlight",
+    "author": "Joy Akpan"
+  },
+  {
+    "title": "The Mechanic",
+    "author": "Koffi Frank"
+  }
+]
```
Comment (Ibizugbe) [inline json/books.json]: It is also impossible to create a book and preserve the data as the same error as shown in the person is being thrown here. Kindly fix it👍.
image

**Review by Ibizugbe — CHANGES_REQUESTED** [review]
Hi @abiola-arobieke  @abiola-arobieke 👋👋, 

tryagain


Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights✅✅✅

- [x] Nice job making some corrections as requested by the previous reviewer👍.


## Required Changes ♻️

- [ ] Due to the errors below, it was also impossible to test the rentals😉. Kindly fix the errors below😉.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by sumairq — APPROVED** [review]
### Hi Team , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Good commit history
✔️Linters are passing
✔️Very professional README
✔️Correct Git flow
✔️Descriptive PR
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**